// unimport/dist/index.mjs
export { b as builtinPresets, c as createUnimport, e as dedupeDtsExports, r as resolveBuiltinPresets, a as resolvePreset, d as scanDirExports, f as scanExports, s as scanFilesFromDir, v as version } from './shared/unimport.4aa8fb3c.mjs';
export { h as addImportToCode, a as dedupeImports, d as defineUnimportPreset, j as excludeRE, f as getMagicString, g as getString, k as importAsRE, m as matchRE, n as normalizeImports, r as resolveIdAbsolute, l as separatorRE, s as stringifyImports, o as stripCommentsAndStrings, t as toExports, i as toImports, c as toTypeDeclarationFile, b as toTypeDeclarationItems, e as toTypeReExports, v as vueTemplateAddon } from './shared/unimport.5eb3d324.mjs';
import 'node:os';
import 'node:fs';
import 'mlly';
import 'pkg-types';
import 'pathe';
import 'node:process';
import 'local-pkg';
import 'node:fs/promises';
import 'fast-glob';
import 'scule';
import 'magic-string';
import 'strip-literal';

async function installGlobalAutoImports(imports, options = {}) {
  const {
    globalObject = globalThis,
    overrides = false
  } = options;
  imports = Array.isArray(imports) ? imports : await imports.getImports();
  await Promise.all(
    imports.map(async (i) => {
      if (i.disabled || i.type)
        return;
      const as = i.as || i.name;
      if (overrides || !(as in globalObject)) {
        const module = await import(i.from);
        globalObject[as] = module[i.name];
      }
    })
  );
  return globalObject;
}

export { installGlobalAutoImports };
